event/listener.php

```diff
@@ -111,6 +112,22 @@ public function do_poll_modification($event)
 	}
 
 	/**
+	 * Modifys the template vars to match the advanced poll settings
+	 *
+	 * @param object $event The event object
+	 * @return void
+	 */
+	public function do_poll_block_modification($event)
+	{
+		if ($event['s_display_results'] && (int) $event['topic_data']['wolfsblvt_poll_show_ordered'])
```

Comment:
This is intentional: poll should not be ordered unless RESULTS are
displayed, if you have not voted, order should be by option id (so that you
are not influenced).  So no, it should not run if you have not voted, or if
the results are hidden.  I will verify the coding anyway, as I have to look
carefully at the conditions.

Ah, AJAX... that is the one bit that I don't know how to make it work, the
JS part that I mentioned.  Any suggestion on how to address this?

Regarding the type casting, ok, no big deal :).

2015-04-07 3:59 GMT+02:00 Wolfsblvt <notifications@github.com>:

> In event/listener.php
> <https://github.com/Wolfsblvt/advancedpolls/pull/17#discussion_r27846860>:
>
> > @@ -111,6 +112,22 @@ public function do_poll_modification($event)
> >  	}
> >
> >  	/**
> > +	 * Modifys the template vars to match the advanced poll settings
> > +	 *
> > +	 * @param object $event The event object
> > +	 * @return void
> > +	 */
> > +	public function do_poll_block_modification($event)
> > +	{
> > +		if ($event['s_display_results'] && (int) $event['topic_data']['wolfsblvt_poll_show_ordered'])
>
> Sure that $event['s_display_results'] is the best idea? I mean if you
> haven't voted, this function will not run. But if you vote then via AJAX,
> the votes are not ordered.
> So why not use the same idea like mine, check for
> isset($topic_data['poll_title']) there?
> Oh, and the int cast is not needed I would say. If you want typecasting,
> (bool) would be the correct here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Wolfsblvt/advancedpolls/pull/17/files#r27846860>.
>